### ISSUE TITLE
Be able to consume CSV

### ISSUE THREAD

**Issue by s3u** [ql-io/ql.io#49]
There is a lot of CSV data out there!

**Comment by s3u** [comment]
I think it is better to make media type sniffing and to-json be pluggable, and let apps initialize engine with different plugins.

**Comment by shimonchayim** [comment]
Good idea. We could also allow another monkey patch function which is responsible for parsing the response and returning engine Json.

**Comment by s3u** [comment]
I meant the connect middleware style, and not patching.

**Comment by shimonchayim** [comment]
Likely going to you ya-csv node package. Comments?

**Comment by s3u** [comment]
Yes.

```
var reader = csv.createCsvFileReader('data.csv', { columnsFromHeader: true });
reader.addListener('data', function(data) {
    // supposing there are so named columns in the source file
    sys.puts(data.col1 + " ... " + data.col2);
});
```

**Comment by shimonchayim** [comment]
Likely will use https://github.com/wdavidw/node-csv-parser

**Comment by shimonchayim** [comment]
Test code:

   var csv = require('csv');

```
var jsonData = [];   
csv()
.from('id,lastname,firstname\r\n82,Preisner,Zbigniew\r\n94,Gainsbourg,Serge',{
    columns: true
})
.transform(function(data){
jsonData.push(data);
console.log(data);
    return data;
})
.on('end', function() {
    console.log(jsonData);
});
```

**Comment by shimonchayim** [comment]
Fixed with pull #167
